### PR TITLE
Local caching/sourcing

### PR DESCRIPTION
**Pull request by shepwalker**
Downloads all files on initial load, tries to default to local copies where available, provides remote when not.

### CODE REVIEW

**Comment by shepwalker** [comment]
Fixed most-to-all of the things.

File: run.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
 # This Python file uses the following encoding: utf-8
 
-from flask import Flask, request
+from flask import Flask, request, send_from_directory, abort, url_for, request
```
Comment (jmduke) [inline run.py]: break out into newlines since it's >#

File: run.py [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
 import requests
 import twilio.twiml
+import os
```
Comment (jmduke) [inline run.py]: organize by import type:

https://www.python.org/dev/peps/pep-0008/#imports

File: run.py [hunk 3]
```diff
@@ -13,21 +16,52 @@
...
+        directory = cache_dir + '/' + emoji_dir
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+            urllib.request.urlretrieve(corgi, directory + "/01.jpg")
+    except:
+        print("Failed on: " + i)
+
+
+@app.route("/emoji/<path:file_name>", methods=['GET'])
+def get_image(file_name):
```
Comment (jmduke) [inline run.py]: add a lil docstring

File: run.py [hunk 4]
```diff
@@ -13,21 +16,52 @@
...
+    full_file_name = cache_dir+"/"+file_name
```
Comment (jmduke) [inline run.py]: spaaaces

**Comment by jmduke** [comment]
Merged!

**Pull request merged**